pbm/pbm_util.go

```diff
@@ -146,3 +152,99 @@ func verifyPropertyValueIsBoolean(propertyValue string, dataType string) (bool,
 	}
 	return val, nil
 }
+
+// ProfileMap contains a map of storage profiles by name.
+type ProfileMap struct {
+	Name    map[string]types.BasePbmProfile
+	Profile []types.BasePbmProfile
+}
+
+// ProfileMap builds a map of storage profiles by name.
+func (c *Client) ProfileMap(ctx context.Context, uid ...string) (*ProfileMap, error) {
+	m := &ProfileMap{Name: make(map[string]types.BasePbmProfile)}
+
+	rtype := types.PbmProfileResourceType{
+		ResourceType: string(types.PbmProfileResourceTypeEnumSTORAGE),
+	}
+
+	category := types.PbmProfileCategoryEnumREQUIREMENT
+
+	ids := make([]types.PbmProfileId, len(uid))
```

Comment:
Nit: Use `var ids []types.PbmProfileId` here and init the slice in the "else" statement to avoid unnecessary allocation when `len(uid) == 0`.

Reply:
I considered that and opted to have 1 less line of code.. but I agree this pattern/practice is better in general, updated.